File: Corpo.Domain/Models/TestExerciseMember.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Corpo.Domain.Models
{
    public class TestExerciseMember
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public TestType TestType { get; set; }
        public int? Minutes { get; set; }
        public int? Seconds { get; set; }
        public string? Video { get; set; }
        public virtual TestMember TestMember { get; set; }
        public int TestMemberId { get; set; }
        public StatusTest Status { get; set; }
        public virtual TestHeartRateExercise TestHeartRateExercise { get; set; }
        public virtual TestRepetitionExercise TestRepetitionExercise { get; set; }
        public virtual TestVideoExercise TestVideoExercise { get; set; }
    }
}

File: Corpo.Web/ClientApp/src/app/domain/test/test-exercise-member.ts
import { StatusTest } from "./test-member";

export class TestExerciseMember {
  id: number;
  name: string;
  protocol: string;
  testType: TestType;
  video: string;
  minutes: number;
  seconds: number;
  status: StatusTest;
  exerciseFmsId: number
}


export enum TestType {
  HeartRate = 1,
  Repetition = 2,
  video = 3
}
